submission-history/src/index.ts

```diff
@@ -0,0 +1,281 @@
+            };
+            let sql = SqlString.format('INSERT INTO submissionTable SET ?', data);
+            let result = await db.query(sql);
+            res.status(200).json({
+                status: "success",
+                message: "Data inserted..."
+            });
+        }
+        catch (error) {
+            console.log("Insert Submimssion Table " + error);
+            res.status(401).json({
+                status: "failed",
+                message: error
+            });
+        }
+    });
+
+    // Gerry's request
+    // most recent first
+    server.get('/getUserSubmissions', (req, res) => {
+        let sql = SqlString.format('SELECT * FROM submissionTable WHERE userId = ? ORDER BY dateSubmitted DESC', [req.query.userId]);
+        let query = db.query(sql, (error, results) => {
+            if (error) {
+                console.log("Select " + error);
+                if (req.query.userId == '') {
+                    res.status(404).json({
+                        status: "failed",
+                        message: "ID is missing"
+                    });
+                }
+                else {
+                    res.status(401).json({
+                    status: "failed",
+                        message: "ID is mismatch"
+                    });
+                }
+            }
+            else {
+                res.status(200).json({
+                    status: "success",
+                    message: "Post fetched...",
+                    data: {
+                        "challengeId": challengeID = results.slice(-10).map(item => item.challengeId),
+                        "challengeName": challengeName = results.slice(-10).map(item => item.challengeName),
+                        "dateSubmitted": dateSubmittedG = results.slice(-10).map(item => item.dateSubmitted.toISOString().slice(0, 19).replace('T', ' ')),
+                        "didAllTestsPass": didAllTestsPass = results.slice(-10).map(item => item.didAllTestsPass)
+                    }
+                    
+                });
+                /**
+                res.render('submissions', {
+                    challengeId: challengeID = results.slice(-10).map(item => item.challengeId),
+                    challengeName: challengeName = results.slice(-10).map(item => item.challengeName),
+                    dateSubmitted: dateSubmittedG = results.slice(-10).map(item => item.dateSubmitted.toISOString().slice(0, 19).replace('T', ' ')),
+                    didAllTestsPass: didAllTestsPass = results.slice(-10).map(item => item.didAllTestsPass)
+                });
+                **/
+            }
+        });
+    });
+
+    // Austin's request 
+    // fastest first
+    server.get('/getChallengeHighscores', (req, res) => {
+        let sql = SqlString.format('SELECT * FROM submissionTable WHERE challengeId = ? AND programmingLanguage = ? ORDER BY executionTime', [req.query.challengeId, req.query.programmingLanguage]);
```

Comment:
You should set a limit for the number of results so you don't need to slice the returned array later.

Reply:
Removed slice() since I already set the limit